### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by trustin**
- Bucket4J 4.10.0 -> 6.2.0
- Curator 4.3.0 -> 5.1.0
  - ZooKeeper 3.5.8 -> 3.6.2
- Dropwizard 2.0.18 -> 2.0.20
- Dropwizard Metrics 4.1.17 -> 4.1.18
- gRPC 1.35.0 -> 1.36.1
- Jackson 2.12.0 -> 2.12.2
- java-jwt 3.12.1 -> 3.14.0
- Jetty 9.4.36 -> 9.4.39
- Micrometer 1.6.3 -> 1.6.5
- Netty 4.1.58 -> 4.1.63
  - Netty TCNative BoringSSL 2.0.36 -> 2.0.38
  - Netty io_uring transport 0.0.3 -> 0.0.5
- OpenSAML 3.4.5 -> 3.4.6
  - Shibboleth java-support 7.5.1 -> 7.5.2
- Reactor 3.4.2 -> 3.4.4
  - Reactor Kotlin extensions 1.1.2 -> 1.1.3
- RxJava 3.0.9 -> 3.0.11, 2.2.20 -> 2.2.21
- ScalaPb 0.10.10 -> 0.11.0
- Spring Boot 2.4.2 -> 2.4.4
- Tomcat 9.0.41 -> 9.0.44, 8.5.61 -> 8.5.64
- Shaded
  - futures-extra 4.2.2 -> 4.3.0
  - Guava 30.1 -> 30.1.1
  - JTools 3.2.0 -> 3.3.0
- Build
  - akka-actor-typed 2.6.11 -> 2.6.13
  - akka-grpc-runtime 1.0.2 -> 1.0.3
  - Checkstyle 8.39 -> 8.41.1
  - Error Prone 2.5.1 -> 2.6.0
  - eureka-client 1.10.11 -> 1.10.13
  - Finagle 20.10.0 -> 21.3.0
  - gax-grpc 1.60.1 -> 1.62.0
  - gradle-node-plugin 2.2.4 -> 3.0.1
  - JMH 1.26 -> 1.29
  - jmh-gradle-plugin 0.5.2 -> 0.5.3
  - json-unit 2.23.0 -> 2.25.0
  - JUnit 5.7.0 -> 5.7.1, 4.13.1 -> 4.13.2
  - Mockito 3.7.7 -> 3.8.0
  - TestNG 7.3.0 -> 7.4.0
- Examples
  - Dagger 2.31.2 -> 2.33
  - Kotlin 1.4.20 -> 1.4.32
    - KotlinX Coroutines 1.4.2 -> 1.4.3
- Updated front-end dependencies, as usual
- Miscellaneous:
  - Added a workaround for JVM crashes.
  - Made all builder methods in `ArmeriaResteasyClientBuilder` return
    itself.
  - Fixed a false-positive warning about version inconsistencies in
    `TransportTypeProvider`.
  - Fixed a bug where an error response is not displayed in docs-client.
  - Simplified JSON parsing and formatting in docs-client.
  - Collect `hs_err_pid*.log` file as well on a build failure.

### CODE REVIEW

**Comment by trustin** [comment]
Some dependencies couldn't be updated, e.g.
- akka-http (getting 415 HTTP status code)
- assertj (thinks we're running in TestNG)
- fastutil (ProGuard fails to trim it)

**Comment by trustin** [comment]
Note that I didn't add the `defect` label in this PR because the fixes here are for unreleased stuff.

**Review by minwoox — APPROVED** [review]
Thank you!

File: .appveyor.report.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ TARBALL="$TARBALL_BASENAME.tar"
 
 msg 'Collecting the test reports ..'
 echo_and_run rm -f "$TARBALL"
-echo_and_run find . -path '*/build/reports/tests' -type d \
+echo_and_run find . '(' -name 'hs_err_*.log' -or -path '*/build/reports/tests' ')' \
```
Comment (ikhoon) [inline .appveyor.report.sh:27]: 👍

**Review by ikhoon — APPROVED** [review]
Thanks! @trustin

File: dependencies.yml [hunk 2]
```diff
@@ -122,18 +125,18 @@ com.google.protobuf:
 
 # Akka is used only for testing in it:grpcweb module.
 com.lightbend.akka.grpc:
-  akka-grpc-runtime_2.13: { version: '1.0.2' }
+  akka-grpc-runtime_2.13: { version: '1.0.0' }
```
[the remark this answers is not in the document]
Comment (trustin) [reply dependencies.yml]: Oops! Upgraded to 1.0.3

**Pull request merged**